addons/sale_stock/models/sale_order.py

```diff
@@ -49,7 +49,7 @@ def _compute_expected_date(self):
         for order in self:
             dates_list = []
             confirm_date = fields.Datetime.from_string(order.confirmation_date if order.state == 'sale' else fields.Datetime.now())
-            for line in order.order_line.filtered(lambda x: x.state != 'cancel'):
+            for line in order.order_line.filtered(lambda x: x.state != 'cancel' and x.product_id.type  in ["product", "consu"]):
```

Comment:
I think the only modification needed is here:
``` python
for line in order.order_line.filtered(lambda x: x.state != 'cancel' and not x._is_delivery()):
```

Nothing prevents setting a `customer_lead` on a service, see:
https://github.com/odoo/odoo/blob/12.0/addons/sale/views/sale_views.xml#L355-L358